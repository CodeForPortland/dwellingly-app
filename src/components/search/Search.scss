@import '../../constants/constants.scss';
.searchContainer {
  position: relative;
  display: flex;
  flex-direction: column;
  padding-top: 1em;
  & .searchContainerInner {
    display: flex;
    flex-direction: column;
    justify-content: center;
    & .searchBar {
      border: 1px solid $color__gray--border;
      border-radius: 25px;
      margin: 0;
      position: relative;
      z-index: 2;
      padding: 0.350rem 1rem;
      width: 100%;
      max-width: 400px;
      &.active {
        opacity: .6;
        letter-spacing: 0.077px;
      }
    }
    & .inline-input__label {
      &.sml-text {
        font-size: .91rem;
      }
    }
    // & .dropdownIcon {
    //   color: $color__gray--medium;
    //   position: absolute;
    //   font-size: .8em;
    //   top: 14px;
    //   right: 12px;
    //   transform: rotate(90deg);
    //   z-index: 3;
    // }
  }
  & .pastSearch {
    width: 100%;
    background-color: white;
    z-index: 1;
    padding: 0.5em;
    letter-spacing: 0.2px;
    text-align: left;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
  }
}
.searchResultsContainer {
  position: absolute;
  z-index: 1;
  top: 16px;
  right: 0;
  left: 0;
  // margin: 0 1em;
  padding: 0 1em;
  padding-top: 38px;
  max-height: 300px;
  max-width: 400px;
  overflow: scroll;
  animation: height 0.1s linear;
  background-color: white;
  @include dialog_shadow;
  border-radius: 25px;
  border-left: 1px solid $color__gray--border;
  border-right: 1px solid $color__gray--border;
  border-bottom: 1px solid $color__gray--border;
  // border-bottom-left-radius: $border-radius--card;
  // border-bottom-right-radius: $border-radius--card;
}
.searchResults {
  width: 100%;
  cursor: pointer;
  padding: 0.5em 0;
  &:hover,
  &:focus {
    background: #ececec;
    outline: none;
  }
  &--multiple {
    padding-left: 20px;
    display: flex;
    align-items: center;
    position: relative;
    &:before {
      position: absolute;
      left: 0px;
      content: "\2610";
      color: $color__black;
      font-size: 1.2em;
    }
    &-active {
      position: relative;
      padding-left: 20px;
      &:before {
        content: "\2611";
        font-size: 1.2em;
        color: $color__black;
        position: absolute;
        left: 0;
      }
    }
  }
}

@keyframes height {
  0% {
    opacity: 0;
    height: 0%;
  }
  50% {
    height: 50%;
  }
  100% {
    opacity: 1;
    height: 100%;
  }
}
