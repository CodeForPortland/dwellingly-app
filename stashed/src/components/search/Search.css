/* V A R I A B L E S */
/* end variables */
/* F U N C T I O N S */
/* end functions */
/* M I X I N S */
/* end mixins */
.searchContainer {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  padding-top: 1em;
}

.searchContainer .searchContainerInner {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.searchContainer .searchContainerInner .searchBar {
  border: 1px solid #dddddd;
  border-radius: 25px;
  margin: 0;
  position: relative;
  z-index: 2;
  padding: 0.350rem 1rem;
  width: 100%;
  max-width: 400px;
}

.searchContainer .searchContainerInner .searchBar.active {
  opacity: .6;
  letter-spacing: 0.077px;
}

.searchContainer .searchContainerInner .inline-input__label.sml-text {
  font-size: .91rem;
}

.searchContainer .searchContainerInner .selectedOptions {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  margin-top: 20px;
}

.searchContainer .searchContainerInner .selectedOptions .selectedPill {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  padding: 3px 10px;
  margin: 0 0.3rem;
  border-radius: 18px;
  background: #eeeeee;
}

.searchContainer .searchContainerInner .selectedOptions .selectedPill:first-child {
  margin-left: 0;
}

.searchContainer .searchContainerInner .selectedOptions .selectedPill .pillClose {
  cursor: pointer;
  font-size: .7em;
  margin-left: 8px;
  margin-top: 1px;
}

.searchContainer .pastSearch {
  width: 100%;
  background-color: white;
  z-index: 1;
  padding: 0.5em;
  letter-spacing: 0.2px;
  text-align: left;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
}

.searchResultsContainer {
  position: absolute;
  z-index: 1;
  top: 16px;
  right: 0;
  left: 0;
  padding-top: 38px;
  max-height: 300px;
  max-width: 400px;
  overflow: scroll;
  -webkit-animation: height 0.1s linear;
          animation: height 0.1s linear;
  background-color: white;
  -webkit-box-shadow: 0 2px 3px rgba(0, 0, 0, 0.1);
          box-shadow: 0 2px 3px rgba(0, 0, 0, 0.1);
  border-radius: 25px;
  border-left: 1px solid #dddddd;
  border-right: 1px solid #dddddd;
  border-bottom: 1px solid #dddddd;
}

.searchResults {
  width: 100%;
  cursor: pointer;
  padding: 0.5em 0.5em;
  font-size: 0.875em;
  color: #858585;
}

.searchResults:hover, .searchResults:focus {
  background: #ececec;
  outline: none;
}

.searchResults--multiple {
  padding-left: 1em;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.searchResults--multiple:before {
  content: "\2610";
  color: #171719;
  font-size: 1.2em;
}

.searchResults--multiple-active {
  padding-left: 1em;
}

.searchResults--multiple-active:before {
  content: "\2611";
  font-size: 1.2em;
  color: #171719;
}

@-webkit-keyframes height {
  0% {
    opacity: 0;
    height: 0%;
  }
  50% {
    height: 50%;
  }
  100% {
    opacity: 1;
    height: 100%;
  }
}

@keyframes height {
  0% {
    opacity: 0;
    height: 0%;
  }
  50% {
    height: 50%;
  }
  100% {
    opacity: 1;
    height: 100%;
  }
}
/*# sourceMappingURL=Search.css.map */