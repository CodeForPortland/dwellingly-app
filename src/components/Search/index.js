import React, {useState} from 'react';
import './search.scss';

//Necessary for search to work: input data, where you want the data output, isFiltered (true/false) location

//Potential Error: I'm using shift() to remove the id of each property before filtering. depending on the data within each search, we may or may not want this.

function Search(props) {
    const { input, outputLocation, isFilteredLocation, setOutputState, setIsFilteredStateFalse } = props;

    const enterSearchHandler = (event) => {
      var keyCode = event.keyCode;
      if (keyCode === 13){
        this.searchProperties();
      }
    };

    const clearSearch = () => {
      props.setIsFilteredStateFalse();
      document.getElementById("searchQueryComponent").value = "";

    };

    const searchProperties = () => {
      let allData = input;
      let output = [];
      let searchQuery = document.getElementById("searchQueryComponent").value.toLowerCase().trim();

       if(searchQuery.length > 0){
          for (var i=0;i < allData.length; i++) {
              let dataPoint = Object.values(allData[i]);
              dataPoint.shift();
              if (dataPoint.toString().toLowerCase().includes(searchQuery)){
                  output.push(allData[i]);
            }
          };


      props.setOutputState(output, true)

      }

  };


    return (
      <div className="search-section">
        <input className="input search is-rounded" id="searchQueryComponent" onKeyDown={enterSearchHandler} placeholder="Search properties by name, address, or property manager"></input>
          <button className="save_button button is-rounded" onClick={searchProperties}type="submit">
             Search
          </button>
          <button className="save_button button is-rounded clearButton" onClick={clearSearch}type="submit">
             Clear Search
          </button>
      </div>
    );
};

export default Search;
