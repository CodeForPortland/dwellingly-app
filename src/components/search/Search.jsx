import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Icon from '../icon/Icon';

import './Search.scss';

class Search extends Component {
  constructor(props) {
    super(props)

    this.handleSearch = this.handleSearch.bind(this);
    this.handleSelection = this.handleSelection.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
    this.handleTab = this.handleTab.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.handleShowOptionsList = this.handleShowOptionsList.bind(this);
    this.handleHideOptionsList = this.handleHideOptionsList.bind(this);
    this.handleSelectionClasses = this.handleSelectionClasses.bind(this);
    this.handleUpdatingOptions = this.handleUpdatingOptions.bind(this);
    // this.handleDeletingSelected = this.handleDeletingSelected.bind(this);

    this.state = {
      searchTerm: '',
      selectedOptions: [],
      focus: null
    }
  }

  handleBlur(event) {
    if(this.node && this.node.contains(event.target)) {
      return
    }
    this.handleFocus(event);
  }

  handleFocus(event) {
    const { focus } = this.state;
    if(focus && this.node && this.node.contains(event.target)) {
      return
    }
    if(focus) {
      this.handleHideOptionsList();
    } else {
      this.handleShowOptionsList();
    }
  }

  handleTab(event) {
    if(event.key !== 'Tab' || event.which !== 9) {
      return
     }
     this.handleBlur(event);
  }

  handleShowOptionsList() {
    document.addEventListener("click", this.handleBlur, false);
    document.addEventListener("keyup", this.handleTab, false);
    this.setState({ focus: true });
  }

  handleHideOptionsList() {
    document.removeEventListener("click", this.handleBlur, false);
    document.removeEventListener("keyup", this.handleTab, false);
    this.setState({ searchTerm: '' });
    this.setState({ focus: false });
  }

  handleUpdatingOptions(data) {
    const newDataObj = {
      id: data.id,
      text: this.props.filterSubset.map(item => data[item]).join(' ')
    };

    if(!this.state.selectedOptions.find(({id}) => id === data.id)) {
      this.setState(prevState => ({
        selectedOptions: [...prevState.selectedOptions, newDataObj]}));
    } else {
      this.setState(prevState => ({
        selectedOptions: prevState.selectedOptions.filter(({id}) => id !== data.id)
      }));
    }
  }

  // handleDeletingSelected(data) {
  //   this.setState(prevState => ({selectedOptions: prevState.selectedOptions.filter(item => item !== data)}))
  // }

  handleSelection(searchedObj) {
    // New Tenant Form Callback
    this.props.onSearchSelection(searchedObj);
    if(this.props.multiple) {
      this.handleUpdatingOptions(searchedObj);
    } else {
      this.handleHideOptionsList();
    }
  }

  handleSelectionClasses(data) {
    if(this.props.multiple && this.state.selectedOptions.find(({id}) => id === data)) {
      return "searchResults--multiple-active"
    }
    if(this.props.multiple) {
      return "searchResults--multiple"
    }
    return "searchResults"
  }

  handleSearch(event) {
    const { target } = event;
    const { value } = target;

    this.setState({
      searchTerm: value
    });
  }

  render() {
    const { id, searchData, multiple, placeholder, filterSubset } = this.props;
    const { focus, searchTerm } = this.state;
    const filterSearch = searchData.filter(data => {
      const dataString = filterSubset.map(item => data[item]).join(' ').toLowerCase();
      return dataString.includes(searchTerm.toLowerCase())
    })
    return (
      <div className="searchContainer">
        <div className="searchContainerInner" ref={node => { this.node = node }}>
        {/* needs class "input inline-input" */}
          <div className="align--left">
            <div>
            <label htmlFor={id}>
              {/* <span className={label.length > 20 ? "inline-input__label sml-text" : "inline-input__label"}>{label}</span> */}
              <input
                type="text"
                id={id}
                className="searchBar"
                placeholder={placeholder}
                onFocus={this.handleFocus}
                onChange={this.handleSearch}
                value={searchTerm} />
            </label>
            {/* <span className="dropdownIcon"><Icon icon="arrowRight" /></span> */}
            </div>
          </div>
          {(searchTerm && focus) && (
            <div className="searchResultsContainer">
              {filterSearch.map(data =>
                <div className="align--left" key={data.id}>
                  <button
                    key={data.id}
                    type="button"
                    aria-label={`searchResult - ${data.name}`}
                    className={this.handleSelectionClasses(data.id)}
                    onClick={() => this.handleSelection(data)}>
                    {filterSubset.map(item => <span>{data[item]}&nbsp;</span>)}
                    {/* !multiple ?
                      <span>{data.name} {data.address}</span> :
                      <span>{data.firstName} {data.lastName}</span>
                    */}
                  </button>
                </div>
              )}
            </div>
          )}
          {multiple && (
            <div className="selectedOptions">
              {this.state.selectedOptions.map(option =>
                <div className="selectedPill" key={option.id}>
                  <span>{option.text}</span>
                  <span className="pillClose" onClick={() => this.handleSelection(option)} role="presentation">
                    <Icon icon="close" />
                  </span>
                </div>
              )}
            </div>
          )}
        </div>
      </div>
    )
  }
}

Search.propTypes = {
  filterSubset: PropTypes.arrayOf(PropTypes.string).isRequired,
  id: PropTypes.string,
  onSearchSelection: PropTypes.func.isRequired,
  searchData: PropTypes.arrayOf(PropTypes.object).isRequired,
  multiple: PropTypes.bool,
  placeholder: PropTypes.string
}
Search.defaultProps = {
  id: undefined,
  multiple: undefined,
  placeholder: undefined
}

export default Search
