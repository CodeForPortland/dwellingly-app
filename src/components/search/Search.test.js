import React from 'react';
import renderer from 'react-test-renderer';
import Search from './Search';

const SearchComponentTest = (props) =>
<Search
{...defaultProps}
{...props}
/>;

describe('Search Component', () => {
  it('correctly renders Search component', () => {
    const SearchComponent = renderer.create(SearchComponentTest).toJSON();
    expect(SearchComponent).toMatchSnapshot();
  });
});
