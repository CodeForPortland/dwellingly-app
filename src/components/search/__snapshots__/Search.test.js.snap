// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component correctly renders Search component 1`] = `null`;
